### PR TITLE
Reuse compiler configuration in quarkus:dev

### LINKED ISSUE #6881: Compilation error on first run in dev mode
**Describe the bug**
After updating to Quarkus 1.2.0.Final sometimes me and my teammate get a compilation error when we run 
`mvn quarkus:dev`

> [ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /some/path/SomeService.java:[107,34] multi-catch statements are not supported in -source 6
  (use -source 7 or higher to enable multi-catch statements)

When we repeat the same command it compiles and starts successfully. Never occurred before on version 1.1.1.Final

**Expected behavior**
Shouldn't get the compile error on executing `mvn quarkus:dev`

**Environment:**
 - Output of `uname -a`: 
`Linux denis 5.3.0-29-generic #31-Ubuntu SMP Fri Jan 17 17:27:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
 - Output of `java -version`: 
```
openjdk version "11.0.5" 2019-10-15
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.5+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.5+10, mixed mode)
```
- Output of `mvn -version`:
```
Apache Maven 3.6.1
Maven home: /usr/share/maven
Java version: 11.0.5, vendor: AdoptOpenJDK, runtime: /home/denis/.sdkman/candidates/java/11.0.5.hs-adpt
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.3.0-29-generic", arch: "amd64", family: "unix"
```
 - Quarkus version or git rev: 
`1.2.0.Final`

pom.xml:
```
<?xml version="1.0"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
		 xmlns="http://maven.apache.org/POM/4.0.0"
		 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
	<modelVersion>4.0.0</modelVersion>
	<groupId>dummy.group</groupId>
	<artifactId>some-service</artifactId>
	<version>1.0.0-SNAPSHOT</version>
	<properties>
		<compiler-plugin.version>3.8.1</compiler-plugin.version>
		<maven.compiler.parameters>true</maven.compiler.parameters>
		<java.version>11</java.version>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
		<quarkus-plugin.version>1.2.0.Final</quarkus-plugin.version>
		<quarkus.platform.artifact-id>quarkus-universe-bom</quarkus.platform.artifact-id>
		<quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
		<quarkus.platform.version>1.2.0.Final</quarkus.platform.version>
		<surefire-plugin.version>2.22.1</surefire-plugin.version>
		<lombok.version>1.18.10</lombok.version>
		<maxmind.geoip2.version>2.13.0</maxmind.geoip2.version>
		<amazon.sdk.version>2.10.48</amazon.sdk.version>
	</properties>
	<dependencyManagement>
		<dependencies>
			<dependency>
				<groupId>${quarkus.platform.group-id}</groupId>
				<artifactId>${quarkus.platform.artifact-id}</artifactId>
				<version>${quarkus.platform.version}</version>
				<type>pom</type>
				<scope>import</scope>
			</dependency>
		</dependencies>
	</dependencyManagement>
	<dependencies>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-resteasy</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-resteasy-jackson</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-smallrye-health</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-smallrye-metrics</artifactId>
		</dependency>
		<dependency>
			<groupId>com.maxmind.geoip2</groupId>
			<artifactId>geoip2</artifactId>
			<version>${maxmind.geoip2.version}</version>
		</dependency>
		<dependency>
			<groupId>software.amazon.awssdk</groupId>
			<artifactId>s3</artifactId>
			<version>${amazon.sdk.version}</version>
		</dependency>
		<dependency>
			<groupId>org.projectlombok</groupId>
			<artifactId>lombok</artifactId>
			<version>${lombok.version}</version>
			<scope>provided</scope>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-junit5</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>io.rest-assured</groupId>
			<artifactId>rest-assured</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>org.mockito</groupId>
			<artifactId>mockito-core</artifactId>
			<scope>test</scope>
		</dependency>
	</dependencies>
	<build>
		<plugins>
			<plugin>
				<groupId>io.quarkus</groupId>
				<artifactId>quarkus-maven-plugin</artifactId>
				<version>${quarkus-plugin.version}</version>
				<executions>
					<execution>
						<goals>
							<goal>build</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
			<plugin>
				<artifactId>maven-compiler-plugin</artifactId>
				<version>${compiler-plugin.version}</version>
				<configuration>
					<source>${java.version}</source>
					<target>${java.version}</target>
					<encoding>${project.build.sourceEncoding}</encoding>
				</configuration>
			</plugin>
			<plugin>
				<artifactId>maven-surefire-plugin</artifactId>
				<version>${surefire-plugin.version}</version>
				<configuration>
					<systemProperties>
						<java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
					</systemProperties>
				</configuration>
			</plugin>
		</plugins>
	</build>
	<profiles>
		<profile>
			<id>native</id>
			<activation>
				<property>
					<name>native</name>
				</property>
			</activation>
			<build>
				<plugins>
					<plugin>
						<artifactId>maven-failsafe-plugin</artifactId>
						<executions>
							<execution>
								<goals>
									<goal>integration-test</goal>
									<goal>verify</goal>
								</goals>
								<configuration>
									<systemProperties>
										<native.image.path>
											${project.build.directory}/${project.build.finalName}-runner
										</native.image.path>
									</systemProperties>
								</configuration>
							</execution>
						</executions>
					</plugin>
				</plugins>
			</build>
			<properties>
				<quarkus.package.type>native</quarkus.package.type>
			</properties>
		</profile>
	</profiles>
</project>
```

### PR DESCRIPTION
**Pull request by gastaldi**
Fixes #6881

### CODE REVIEW

**Comment by gastaldi** [comment]
@DenisKipchakbaev can you build this PR locally and confirm that it this works for you?

**Comment by deniswed** [comment]
> @DenisKipchakbaev can you build this PR locally and confirm that it works for you?

yes, I'll try it. Thanks!

**Comment by DenisKipchakbaev** [comment]
I've tried it and it seems OK! 
What did I do:
- Cloned your fork:
`git clone git@github.com:gastaldi/quarkus.git`
- Built and installed it on local maven repository:
`mvn clean install -DskipTests`
- Changed pom.xml, put 999-VERSION for quarkus-maven-plugin and platform:
```
<quarkus-plugin.version>999-SNAPSHOT</quarkus-plugin.version>
<quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
<quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
<quarkus.platform.version>999-SNAPSHOT</quarkus.platform.version> 
```
Also used quarkus-bom instead of quarkus-universe-bom, because it's not building with this version.

- Run `mvn clean quarkus:dev` several times and every time it started successfully.

Seems OK to me

**Comment by gastaldi** [comment]
@DenisKipchakbaev yes, thank you very much for verifying. 

BTW you can also use `hub` from hub.github.com and do a `hub pr checkout 6894` to checkout and build the PR locally.

**Comment by wedamartinez** [comment]
I also did the same and it is working for me also. Thanks @gastaldi!

**Comment by gastaldi** [comment]
Thank you @wedamartinez! @gsmet feel free to review the PR when possible

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java [hunk 1]
```diff
@@ -266,13 +266,18 @@ public void execute() throws MojoFailureException, MojoExecutionException {
             if (plugin == null) {
                 throw new MojoExecutionException("Failed to locate " + key + " among the project plugins");
             }
+            Xpp3Dom configuration = (Xpp3Dom) plugin.getConfiguration();
+            if (configuration == null) {
+                // TODO: Set sources and target to supported version
```
Comment (gsmet) [inline devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java]: Should we keep this TODO or do it right away?

I think I would go with 8 for now and upgrade to 11 as soon as we drop 8.

WDYT?

**Comment by gastaldi** [comment]
Now that I think better, the best is to leave as is and remove the todo
message.

Setting an explicit source/target will have a different behavior than using
mvn compile,therefore it is correct to assume an empty configuration here.
Need to check on multimodule projects though

Em Qui, 30 de jan de 2020 16:17, Guillaume Smet <notifications@github.com>
escreveu:

**Comment by geoand** [comment]
Nice work!

**Comment by gsmet** [comment]
OK, let's try to get this in by tomorrow. I would really like to have that one in the Alpha1 we will release early next week.

**Comment by gastaldi** [comment]
I removed the TODO message. It should be ok now

**Review by geoand — APPROVED** [review]

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Woot!

**Comment by DenisKipchakbaev** [comment]
Thank you guys for a great job!

**Pull request merged**